### DISCUSSION TITLE
TSC Minutes October/November 2020

### DISCUSSION THREAD

**Discussion by lgirdwood** [thesofproject/sof#7387] [Team Posts]
### Attendees
Present: @lgirdwood @plbossart @mwasko @beatabaranowska @harshapriya-n @cujomalainey @dgreid @dbaluta @bzhg 
Apologies: @cwakeland 

### Agenda
**1) 1.6 release.** 
Release candidates provides valuable community bug reports that have now been resolved. Some PRs still to merge then v1.6 final can be validated before release. 

Problems with mixer and demux due to uncertainties around how these components integrate within topology and how they react to multiple user configurations. Needs a set of rules for pipeline and endpoints. Agreed to use topology to define configurations for mixer and mux components. 

- [ ] Create FEATURE for pipeline and endpoint rules. - @lgirdwood 
- [ ] Create topology configurations for mixer and demux - **TBD**

**2) Zephyr Update.**
Zephyr v2.5 and SOF version v1.6 will have native support for each other on the Intel CAVS1.5 platform. Zephyr v2.6 and SOF v1.7 will support any Zephyr target (which will include all Intel CAVS platforms). Zephyr audio related opens identified for trace, scheduling, and SMP which will merge post v2.6.

**3) SOF v1.7 release.**
Release planned for December or January 2021. Significant features will be codec adapter and possibly multi band DRC time permitting (but trending for v1.8).

- [ ] Tag PRs and Features for v1.7 - @lgirdwood 
- [ ] Upstream compressed PCM support in SOF Linux driver - @dbaluta 

**4) Developer tooling.**
Intel announced there would be tooling to generate ALSA conf from XML file. 
Also gathered feedback that there is the problem with topology generation. Understood the issue and would make the research what could be done.
More details will follow in early 2021.

**5) SOF Linux Driver.**
Mainly cleanups and IPC level abstraction expected in the near future. Aim is to increase modularity with code base to have separate drivers for audio, trace, probes, mediator i.e any FW feature. This relies on Aux bus merging into 5.11 (hopefully) or 5.12.

Mediator work gradually upstreaming as consensus is being reached in virtio mailing list for upstreaming the infrastructure required by the SOF mediator and by other virtio drivers. Linaro are now at v4 of this patch.

**6) Improvements to ALSA conf.**
It was agreed to deprecate M4 due to lack of verification, validation and high learning curve and replace with an improved ALSA conf. This has already been aligned with ALSA conf maintainer.

The new ALSA conf should be simple to understand and be compatible with existing conf. It should take all the existing M4 features like object _reuse_ and data _inclusion_ and be simple to map with the new XML. See https://github.com/thesofproject/sof/pull/3211

**7) Fuzzing.**
Fuzzing going well with a lots of bugs being fixed in the testbench and IPC. Close to be able to run open fuzzing.

**8) Static Analysis**
It was agreed to add some high level static analysis to CI in a phased approach and to have a more comprehensive low level analysis run on a monthly basis.

- [ ] Define the high level static analysis tool and integrate into CI. - @cujomalainey 
- [ ] Define the low level and comprehensive static analysis tool and do 1st run (TBD - how do we report and action the results ??). - @cujomalainey 
 
**9) GSoC Replacement of M4**
@dbaluta This rolls into item 4 and 6, so most of the work should be upstream by next summer. It may be more targeted to provide a visual interface on top of conf ??

**10) Microcontroller**
Running SOF driver via a microcontroller with FW running on DSP. Driver code is dual licensed so can run on other OSes. Missing topology parser so this has to come from testbench.

- [ ] Create feature to complete testbench topology parser support of UUID - @lgirdwood 

**11) Outreachy**
This years outreachy went very well with many improvements and fixes upstreamed. SOF will participate in outreachy in 2021.

- [ ] Participate in Outreachy 2021 - @dbaluta 

### Next Meeting
Next meeting will be in December 2020 before the Christmas holidays. Date TBD.

- [ ]  Create new meeting for TSC December - @lgirdwood

**Comment by beatabaranowska** [comment]
Regarding Point 4: Currently there are no plans to opensource GUI tool. But we are thinking how to make creating topology easier.

**Comment by cujomalainey** [comment]
But the binaries will still be available correct? And will it work for configuring all platforms or just intel?

**Comment by beatabaranowska** [comment]
Exactly which binaries do you mean?

**Comment by cujomalainey** [comment]
The binaries for the gui tool

**Comment by beatabaranowska** [comment]
What I know currently there is no plan for gui tool binaries also.

**Comment by lgirdwood** [comment]
@beatabaranowska could you edit the minutes at section 4 and provide more context about the UI tool, e.g. supported platforms, whether it will be open source, whether it will be released as binary executable (free to community or free to customers). This way it's clear for everyone and on the record. Thanks.

**Comment by beatabaranowska** [comment]
Edited. And once again :)  there is no plan to opensource current UI tool binaries. To be discussed about possibilities in Q1 2021.

**Comment by plbossart** [comment]
@beatabaranowska we can't force people to use a tool and neither provide source nor binaries. What am I missing?

**Comment by beatabaranowska** [comment]
You are right. Edited once again. Hope now it is clear for everyone. Let's gather feedback on next meeting and explain any doubts if exists.
